### PR TITLE
[1.20.5] Refactor PacketDistributor to static helper methods

### LINKED ISSUE #801: [1.20.5] Refactor `PacketDistributor` to static helper methods
This is a note for a future proposed change after completion of #800. No additional changes are being added to that PR due to its size.

Currently, `PacketDistributor` holds many `PacketDistributor` instances, which produce `PacketTarget`s that then are capable of sending payloads.  This has two issues:
1. The system is indirect. The PacketDistributor dispatches to a static method which produces a `Consumer<Packet<?>>`, which is then fed into a `PacketTarget` that users then use to send a payload.
  a. This makes it hard for users to understand the work done by  the distributors.
2. The design of `PacketDistributor` enforces use of a single argument, without having adequate handling for no-argument or multiple-argument targets.

Reducing the class to a series of static helper methods make the usage simpler, and streamlines the additions of new targets.

As an example, `PacketDistributor.PLAYER` could be transformed to the following method:

```java
/**
 * Sends one or more payloads to a specific player.
 * Multiple payloads sent at once will be sent via as a bundle packet.
 */
public static void sendTo(Player player, CustomPacketPayload... payloads) {
    ((ServerPlayer) player).connection.send(toVanillaPacket(payloads));
}
```

Overloads can also be added that take a `Packet<?>` to maintain the functionality offered by `PacketTarget`.  If there are sufficient consumers of `PacketTarget`, or others that wish to maintain the current API style, it is possible to provide the static methods on top of `PacketDistributor`. The drawback is that this will not alleviate concern (1).

### PR DESCRIPTION
**Pull request by Spinoscythe**
TODO: Abstain from omitting the packetflow info
Refactor PacketDistributor to static helper methods.
Fixes #801

### CODE REVIEW

**Comment by Shadows-of-Fire** [comment]
The `Consumer<Packet<?>>` functions should be inlined into the actual helper methods.

Additionally, information about the packetflow needs to be retained and checked as it was previously (the current revision omits it entirely).

File: src/main/java/net/neoforged/neoforge/network/PacketDistributor.java [hunk 1]
```diff
@@ -28,59 +28,66 @@
...
+    public static void sendToTrackingChunk(@Nullable LevelChunk levelChunk, CustomPacketPayload... payloads) {
+        if (levelChunk != null) {
+            trackingChunk(levelChunk).accept(toVanillaPacket(payloads));
+        }
+    }
+
+    private static Packet<?> toVanillaPacket(CustomPacketPayload... payloads) {
+        ClientboundCustomPayloadPacket vanillaPacket = null;
+        for (var p : payloads) {
+            vanillaPacket = new ClientboundCustomPayloadPacket(p);
```
Comment (dhyces) [inline src/main/java/net/neoforged/neoforge/network/PacketDistributor.java]: What in the world is this

**Comment by Spinoscythe** [comment]
I'm running out of ideas on how to abstain from omitting the packet flow without the method returning a type instead of void

**Comment by Spinoscythe** [comment]
I don't think I'm the right guy to do this. srry.

**Pull request closed without merging**